crates/pathfinder/src/sync/p2p/headers.rs

```diff
@@ -182,6 +187,8 @@ pub(super) async fn persist(
             .context("Persisting block header")?;
             tx.insert_signature(header.number, signature)
                 .context("Persisting block signature")?;
+            tx.insert_state_update_stats(header.number, state_update_stats)
+                .context("Persisting state update stats")?;
```

Comment:
I need to think about this :thinking:

Reply:
These counts are re-used right now, ie. they are fetched from the DB in sync_handlers, which ofc requires them being populated from genesis.

But because p2p is still experimental and behind a feature flag I wanted to avoid migrating existing DBs by recalculating the counters for past blocks, because the assumption is that if you're testing P2P some of the nodes (ie. proxy nodes) will be syncing from the gateway from genesis. This means that one part is missing - filling the counts when syncing from the gateway.

I can merge the counts into the headers table. I'm not so sure about the migration actually recalculating the counts for past blocks. Wdyt?